content/platforms/kubernetes/db-controller.md

```diff
@@ -0,0 +1,274 @@
+
+```sh
+kubectl get redb/mydb -o jsonpath="{.spec.databaseSecretName}"
+```
+
+This secret contains:
+
+- The database port (port)
+- The database service name (service_name)
+- The database password for authenticating (password)
+
+The database controller maintains these connection values in the secret.
+An application can use this secret in a variety of ways.
+A simple way is to map them to environment variables in a deployment pod.
+
+For example, we can add the connection parameters as environment variables to deploy a guestbook demonstration application and have it connect to our database:
+
+```yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: guestbook
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: guestbook
+      name: guestbook
+  template:
+    metadata:
+      labels:
+        app: guestbook
+        name: guestbook
+    spec:
+      containers:
+        - name: guestbook
+          image: roeyredislabs/guestbook:latest
+          imagePullPolicy: Always
+          env:
+          - name: REDIS_PORT
+            valueFrom:
+              secretKeyRef:
+                name: redb-mydb
+                key: port
+          - name: REDIS_HOST
+            valueFrom:
+              secretKeyRef:
+                name: redb-mydb
+                key: service_name
+          - name: REDIS_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: redb-mydb
+                key: password
+          ports:
+            - name: guestbook
+              containerPort: 80
+
+```
+
+Then, we can forward the application pod (the name is specific to your deployment):
+
+```sh
+kubectl port-forward guestbook-667fcbf6f6-gztjv 8080:80
+```
+
+Browse to `http://localhost:8080/` to view the demonstration.
+
+
+## Options for Databases
+
+The database options are specified in the `spec` section of the database custom resource.
+These options include options that you can change and options that are created by the controller for applications or developers. For example, the database connection information is stored in a secret.
+
+### `databaseSecretName`
+
+A string containing the name of a secret that contains the desired database password.
+
+The default value is the database name (`metadata.name`) with "redb-" as a prefix.
+
+To create your own database password in the secret, run:
+
+```sh
+ kubectl create secret generic mydb --from-literal=password=<password>
+ ```
+
+where <password> is your desired password in plain text.
+
+In the database custom resource, the value is just secret name. In this example,
+the name is `mydb`:
+
+ ```yaml
+ databaseSecretName: mydb
+ ```
+
+When the database is created, the secret is updated to include the port and service name for the database
+but the password will remain the same.
+
+### `enforceClientAuthentication`
+
+A boolean that indicates whether client authentication should be enforced (default: `true`).
```

Comment:
Please add a reference to this: http://localhost:1313/rs/administering/designing-production/security/client-connections/?s=client%20authentication#configuration-of-the-database (e.g. "learn more about the implication of client authentication")

Reply:
Done